test/enzyme.jl

```diff
@@ -0,0 +1,33 @@
+using Test
+using Enzyme
+using KernelAbstractions
+
+@kernel function square!(A)
+    I = @index(Global, Linear)
+    @inbounds A[I] *= A[I]
+end
+
+function caller(A, backend)
+    kernel = square!(backend)
+    kernel(A, ndrange=size(A))
+    synchronize(backend)
+end
+
+function enzyme_testsuite(backend, ArrayT)
+    @testset "kernels" begin
+        A = ArrayT{Float64}(undef, 64)
+        A .= (1:1:64)
+        dA = ArrayT{Float64}(undef, 64)
+        dA .= 1
+
+        Enzyme.autodiff(Reverse, caller, Duplicated(A, dA), Const(backend()))
```

Comment:
So this test would not pass on GPU backends?

We will need to do feature detection then.

Reply:
Yeah